lib/httpRequestBuilder.js

```diff
@@ -37,7 +37,7 @@ function requestBuilder (defaults) {
     const headers = reqData.headers
     const body = reqData.body
 
-    let host = reqData.host
+    let host = reqData.headers.host || reqData.host
```

Comment:
This does seem to fail when the header is given as `-H Host=value` on the CLI, e.g. it does not work case-insensitively like HTTP headers are supposed to be.

Reply:
wow, this was a long time ago, would you open a fresh issue or a PR to fix?

Reply:
https://github.com/mcollina/autocannon/pull/463

Reply:
I use
```bash
autocannon -H "host=xxx.com" "http://127.0.0.1:8000"
```
it seems that it just replace 127.0.0.1 with xxx.com in url which is not what i intend

```
Request {
  bodyUsed: false,
  headers: Headers { connection: "keep-alive", host: "xxx.com, xxx.com" },
  method: "GET",
  redirect: undefined,
  url: "http://xxx.com/"
}
```

Deno serve
```ts
Deno.serve((req) => {
  console.log(req)
  return new Response('a')
})
```